mythtv/libs/libmyth/mythcontext.cpp

```diff
@@ -1242,8 +1248,8 @@ int MythContextPrivate::UPnPautoconf(const std::chrono::milliseconds milliSecond
 bool MythContextPrivate::DefaultUPnP(QString& Error)
 {
     static const QString loc = "DefaultUPnP() - ";
-    QString pin = m_pConfig->GetValue(kDefaultPIN, "");
-    QString usn = m_pConfig->GetValue(kDefaultUSN, "");
+    QString pin = m_pConfig->GetValue(kDefaultPIN, QString("")); // not saved
+    QString usn = m_pConfig->GetValue(kDefaultUSN, QString("")); // not saved
```

Comment:
These are potentially saved by https://github.com/MythTV/mythtv/blob/befa034b04d8449e0b1d41b0a28905cd9847dc37/mythtv/libs/libmyth/backendselect.cpp#L109-L119

TODO: rewrite this to eliminate pointer-y global state.  Because of how XmlCongifuration::Save() works, don't keep an object alive.

Instead:
1.  create a new object (on the stack)
2. Read whatever.
3. If writing, set (or clear) values, immediately save, and destroy the object.